sdk/eventhub/Azure.Messaging.EventHubs/samples/Program.cs

```diff
@@ -137,6 +381,34 @@ private static void DisplayHelp()
             Console.WriteLine();
         }
 
+        /// <summary>
+        ///   Reads the selection of the application's user from the
+        ///   console for samples.
+        /// </summary>
+        ///
+        /// <param name="samples">The available samples.</param>
+        ///
+        /// <returns>The validated selection that was made.</returns>
+        ///
+        private static int? ReadSelection(IReadOnlyList<IEventHubsSample> samples)
```

Comment:
You could collapse this down into one method if you made it generic.   Something like:

```csharp
private static int? ReadSelection<TSampleCategory>(IReadOnlyList<TSampleCategory> samples) => samples switch
{
    IReadOnlyList<IEventHubsSample> samples => ReadSelection(samples.Count + extraOptionsForSamples.Length),
    IReadOnlyList<IEventHubsIdentitySample> samples => return ReadSelection(samples.Count + extraOptionsForIdentitySamples.Length),
    _ => throw new InvalidArgumentException()
};
```